### DISCUSSION TITLE
Late initiation

### DISCUSSION THREAD

**Discussion by robfordww** [jackc/pgx#1341] [Q&A]
I keep wondering if this is safe:

```
type connection struct {
      p pgxpool.Pool  // note not a pointer
}
```
 then later on, after  passing a pointer to the connection struct around, but before any usage, initialize it with 
```
c.p = *pgxpool.Connect(context.Background(), "....")
where c is the pointer to the connection struct above
```
this obviously leads to a temporary "copy" of the pool, which feels somewhat unsafe given its internal complexity.  But as long as `c.p` has not been used, is it safe?

**Comment by jackc** [comment]
I don't know anything off the top of my head that would for sure be *unsafe*... but I wouldn't rely on it being safe either.
